firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpStream.java

```diff
@@ -0,0 +1,341 @@
+// Copyright 2022 Google LLC
```

Comment:
I'm a bit confused about the separation between `ConfigRealtimeHttpClient` and `ConfigRealtimeHttpStream` since `ConfigRealtimeHttpClient` still maintains a reference to the `HttpUrlConnection` and the business logic is split between the two classes. I think it'd be easier to read all the HTTP connection logic in one place. Thoughts?

Reply:
In ConfigRealtimeHttpClient there's actually no reference to HttpUrlConnection; it's a reference to the FutureTask that is returned after submitting the ConfigRealtimeHttpStream thread to the executor so that we can know if we have a thread running or not. The reasons why I want to keep the Http work separated into it's own class are so it's centralized in a place where there's no other type of work going on and to make it more thread safe since that background thread will be only confined to the ConfigRealtimeHttpStream class and it won't touch any other classes/variables